pyramid/static.py

```diff
@@ -284,3 +204,83 @@ def __init__(self, token, param='x'):
 
     def tokenize(self, pathspec):
         return self._token
+
+class ManifestCacheBuster(object):
+    """
+    An implementation of :class:`~pyramid.interfaces.ICacheBuster` which
+    uses a supplied manifest file to map an asset path to a cache-busted
+    version of the path.
+
+    The file is expected to conform to the following simple JSON format:
+
+    .. code-block:: json
+
+       {
+           "css/main.css": "css/main-678b7c80.css",
+           "images/background.png": "images/background-a8169106.png",
+       }
+
+    Specifically, it is a JSON-serialized dictionary where the keys are the
+    source asset paths used in calls to
+    :meth:`~pyramid.request.Request.static_url. For example::
+
+    .. code-block:: python
+
+       >>> request.static_url('myapp:static/css/main.css')
+       "http://www.example.com/static/css/main-678b7c80.css"
+
+    If a path is not found in the manifest it will pass through unchanged.
+
+    If ``reload`` is ``True`` then the manifest file will be reloaded when
+    changed. It is not recommended to leave this enabled in production.
+
+    If the manifest file cannot be found on disk it will be treated as
+    an empty mapping unless ``reload`` is ``False``.
+
+    The default implementation assumes the requested (possibly cache-busted)
+    path is the actual filename on disk. Subclasses may override the ``match``
+    method to alter this behavior. For example, to strip the cache busting
+    token from the path.
+
+    .. versionadded:: 1.6
+    """
+    exists = staticmethod(exists) # testing
+    getmtime = staticmethod(getmtime) # testing
+
+    def __init__(self, manifest_path, reload=False):
+        self.manifest_path = manifest_path
+        self.reload = reload
+
+        self._mtime = None
+        if not reload:
+            self._manifest = self.parse_manifest()
+
+    def parse_manifest(self):
+        """
+        Return a mapping parsed from the ``manifest_path``.
+
+        Subclasses may override this method to use something other than
+        ``json.loads``.
```

Comment:
This method could also be used not only to switch from something other than ``json.loads()`` but also to coerce the underlying manifest into a dictionary in the correct format for ``ManifestCacheBuster``. Might be worth documenting that as well?